File: HiveMind.Server/Entities/Schedule.cs
namespace HiveMind.Server.Entities;

public class Schedule
{
    public int ScheduleId { get; set; }
    public string ScheduleName { get; set; } = string.Empty;
    public int ChannelId { get; set; }
    public TimeOnly StartTime { get; set; }
    public Channel? Channel { get; set; }
    public ICollection<ScheduleItem>? ScheduleItems { get; set; }
}

File: hivemind.client/src/Types/Schedule.ts
export interface ScheduleItem {
    scheduleItemId: number;
    name: string;
    index: number;
    type: string;
    queries: QueryScheduleItem[];
}

export interface QueryScheduleItem {
    queryScheduleItemId: number;
    queryId: number;
    scheduleItemId: number;
    queryType: number;
    index: number;
    padTo: number;
    playCount: number;
    playDuration: number;
    playoutType: number;
}

export interface Schedule {
    scheduleId: number;
    scheduleName: string;
    channelId: number | null;
    startTime: string;
    scheduleItems: ScheduleItem[];
}